asyncgit/src/sync/rebase.rs

```
@@ -0,0 +1,143 @@
+        Err(e) => {
+            if let Ok(mut rebase) = repo.open_rebase(None) {
+                if let Some(cur_branch) = cur_branch_ref {
+                    rebase.abort()?;
+                    repo.set_head(&cur_branch)?;
+                    repo.checkout_head(None)?;
+                }
+            }
+            Err(e)
+        }
+    }
+}
+
+/// Changes the commit message of a commit with a specified oid
+///
+/// While this function is most commonly associated with doing a
+/// reword opperation in an interactive rebase, that is not how it
+/// is implimented in git2rs
+///
+/// This is dangerous if this errors, as the head will be detached so this should
+/// always be wrapped by another function which aborts the rebase if something goes worng
+pub fn reword(
+    repo_path: &str,
+    commit_oid: Oid,
+    message: &str,
+) -> Result<()> {
+    let repo = utils::repo(repo_path)?;
+    let sig = signature_allow_undefined_name(&repo)?;
+
+    let parent_commit_oid = if let Ok(parent_commit) =
+        repo.find_commit(commit_oid)?.parent(0)
+    {
+        Some(parent_commit.id())
+    } else {
+        None
+    };
+
+    let commit_to_change = if let Some(pc_oid) = parent_commit_oid {
+        // Need to start at one previous to the commit, so
+        // next point to the actual commit we want to change
+        repo.find_annotated_commit(pc_oid)?
+    } else {
+        return Err(Error::NoParent);
+        // Would the below work?
+        // repo.find_annotated_commit(commit_oid)?
+    };
+    let mut top_branch_commit = None;
+    let mut cur_branch_ref = None;
+    let mut cur_branch_name = None;
+
+    // Find the head branch
+    for b in repo.branches(None)? {
+        let branch = b?.0;
+        if branch.is_head() {
+            cur_branch_ref = Some(String::from(
+                branch
+                    .get()
+                    .name()
+                    .expect("Branch name is not valid utf8"),
```

why panic instead of returning the error using `?`? We have `Utf8Error` for this in error